### PR TITLE
Convert QOP constants into an enum.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

**Review by michael-o — CHANGES_REQUESTED** [review]
If at all the class should be called QualityOfProtection and the QOP_ prefix is redundant.

**Comment by arturobernalg** [comment]
> If at all the class should be called QualityOfProtection and the QOP_ prefix is redundant.

HI @michael-o 
you're right. changed.

TY

File: httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/DigestScheme.java [hunk 1]
```diff
@@ -100,10 +100,12 @@
...
     };
 
-    private static final int QOP_UNKNOWN = -1;
-    private static final int QOP_MISSING = 0;
-    private static final int QOP_AUTH_INT = 1;
-    private static final int QOP_AUTH = 2;
+    /**
+     * Represent the possible values of quality of protection.
+     */
+    private enum QualityProtection {
```
Comment (michael-o) [inline httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/DigestScheme.java:106]: `QualityOfProtection`

**Review by michael-o — CHANGES_REQUESTED** [review]

File: httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/DigestScheme.java (hunk 1 above)
Comment (arturobernalg) [reply httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/DigestScheme.java]: Hi @michael-o 
Changed.
YU

**Review by michael-o — APPROVED** [review]

**Pull request merged**